hw/ip/otbn/dv/model/otbn_model.cc

```diff
@@ -761,3 +769,9 @@ void otbn_take_loop_warps(OtbnModel *model, OtbnMemUtil *memutil) {
   assert(model && memutil);
   model->take_loop_warps(*memutil);
 }
+
+void otbn_model_set_keymgr_value(OtbnModel *model, svLogicVecVal *key0,
+                                 svLogicVecVal *key1, bool valid) {
```

Comment:
The DPI-C layer passes a `bit` as an `unsigned char`. See H.7.4 in the IEEE spec.

Reply:
This seems to have turned back to a bool. Please make it match the DPI-C ABI. Since `valid` is a `bit` in the SystemVerilog declaration, it is passed as an unsigned char. Convert it from `unsigned char` to `bool` in `OtbnModel::set_keymgr_value` by checking that it's either 0 or 1 and then the boolean will be something like `valid != 0`.